internal/pkg/service/stream/storage/statistics/repository/stats_repo.go

```diff
@@ -0,0 +1,92 @@
+// Package repository provides database operations for sink storage statistics.
+//
+// # Data Model
+//
+// Statistics are stored in the etcd database as [statistics.Value] under the following key format:
+//
+//	storage/stats/<LEVEL:local>/<SLICE_KEY:PROJECT_ID/SOURCE_ID/SINK_ID/FILE_ID/VOLUME_ID/SLICE_ID>/value
+//
+// Statistics are stored at the slice level, which represents the smallest unit.
+//
+// To obtain statistics for a parent object such as a file, sink, source, or project,
+// simply sum up all the values under the corresponding prefix.
+//
+//	For a project:   storage/stats/<LEVEL>/<PROJECT_ID>/
+//	For a source:    storage/stats/<LEVEL>/<PROJECT_ID>/<SOURCE_ID>/
+//	For an sink:     storage/stats/<LEVEL>/<PROJECT_ID>/<SOURCE_ID>/<SINK_ID>/
+//	For a file:      storage/stats/<LEVEL>/<PROJECT_ID>/<SOURCE_ID>/<SINK_ID>/<FILE_ID>/
+//	For a slice:     storage/stats/<LEVEL>/<PROJECT_ID>/<SOURCE_ID>/<SINK_ID>/<FILE_ID>/<VOLUME_ID>/<SLICE_ID>
+//
+// # Rollup
+//
+// Over time, files and slices expire and are deleted during the storage Cleanup operation.
+// Their statistics in the "local" and "staging" levels are deleted.
+// Statistics in the "target" level are summarized under the key:
+//
+//	stats/<LEVEL:target>/<SINK_KEY:PROJECT_ID/SOURCE_ID/SINK_ID>/_sum
+package repository
+
+import (
+	"fmt"
+
+	etcd "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/serde"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/plugin"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level"
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+)
+
+const (
+	// putMaxStatsPerTxn defines maximum number of keys per transaction when updating database values.
+	putMaxStatsPerTxn = 100
```

Comment:
How do we know that this value is correct? What if we face some issue according number of keys per transaction. Should we just increase this number?

Reply:
It must be bellow etcd `--max-txn-ops '128'` (default value). 

We'll probably set it to 1000 in etcd just to be sure.

100 is a reasonable value to start with, when there will be a lot of transactions, you can try 1000, what will be the impact.

Reply:
We have K6 E2e benchmarks, there we could see that it is some big % of all transactions, CPU usage, etc.

But this is not a critical part, because those statistics are sent every 1s.